sdk_container/src/third_party/coreos-overlay/coreos/config/env/sys-apps/systemd

```diff
@@ -1,13 +1,437 @@
+                        this_unit_names_to_use=( ${unit_name} )
+                        ;;
+                    'tn') # 2.
+                        this_unit_names_to_use=( ${instances[@]/#/"${unit_name%.*}"} )
+                        this_unit_names_to_use=( ${this_unit_names_to_use[@]/%/".${unit_name##*.}"} )
+                        ;;
+                    'nt') # 3.
+                        die "Case 3. not implemented, need to acquire default instance of the related unit, I was too lazy to implement that; if you have hit this case, congratulations - you get to implement it (debug info: unit name: ${unit_name}, related unit name: ${related_unit_name}, relation: ${related_unit_type}, scope: ${scope})"
+                        ;;
+                    *)
+                        die "Should not happen"
+                        ;;
+                esac
+
+                # this_unit_names_to_use (getty@tty1.service) can be
+                # used for symlink name, but not for the target - we
+                # want the symlink to point to the actual file (like
+                # getty@.service)
+
+                do_by_dir=''
+                tmp_array=()
+                case ${related_unit_type} in
+                    'Alias')
+                        # put paths in array, so I can quote them nicely for einfo
+                        tmp_array+=(
+                            "${systemd_dir}/${scope}/${related_unit_name}"
+                            "${unit_name}"
+                        )
+                        ;;
+                    'WantedBy') do_by_dir='wants';;
+                    'RequiredBy') do_by_dir='requires';;
+                    'UpheldBy') do_by_dir='upholds';;
+                    *) die "Should not happen";;
+                esac
+                if [[ -n ${do_by_dir} ]]; then
+                    dodir "${systemd_dir}/${scope}/${related_unit_name}.${do_by_dir}"
+                    for this_unit_name_to_use in "${this_unit_names_to_use[@]}"; do
+                        # put paths in array, so I can quote them nicely for einfo
+                        tmp_array+=(
+                            "${systemd_dir}/${scope}/${related_unit_name}.${do_by_dir}/${this_unit_name_to_use}"
+                            "../${unit_name}"
+                        )
+                    done
+                fi
+                while [[ ${#tmp_array[@]} -gt 0 ]]; do
+                    dosym "${tmp_array[1]}" "${tmp_array[0]}"
+                    einfo "Created symlink ${tmp_array[0]@Q} -> ${tmp_array[1]@Q}"
+                    tmp_array=( "${tmp_array[@]:2}" )
+                done
+            done
+        done
+    done <"${preset_file}"
+}
+
+cros_post_src_install_flatcar_stuff() {
+    # We provide our own systemd-user config file in baselayout.
+    #
+    # This one is installed by systemd build system regardless of
+    # USE=pam (the ebuild ought to pass -Dpamconfdir=no to disable the
+    # installation).
```

Comment:
So why don't we do that?

Reply:
I think I'll file some PR to Gentoo about it. Just need to think what to do with pam files. systemd from 247 started installing its pam config files into /usr/lib/pam.d, whereas Gentoo, with `newpamd` or `dopamd` installs stuff at `/etc/pam.d`. So, if we build systemd with `USE=-pam`, we should pass `-Dpamconfdir=no` to meson. If we build with `USE=pam` then:

- Should we remove pam files in /usr/lib/pam.d?
- Should ebuild be installing pam files in /usr/lib/pam.d or still in /etc/pam.d?

Also there is a new config file systemd-run0 that does not have its Gentoo counterpart.